integration/singularity/reader.go

```diff
@@ -0,0 +1,105 @@
+package singularity
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/data-preservation-programs/singularity/client/swagger/http/file"
+	"github.com/gotidy/ptr"
+)
+
+// io.ReadSeekCloser implementation that reads from remote singularity
+type SingularityReader struct {
+	store  *SingularityStore
```

Comment:
it looks like you are only using the client, so I would pass that rather than the whole store